src/ArgParse.jl

```diff
@@ -1,10 +1,9 @@
-isdefined(:OptionsMod) || require("Options")
+isdefined(:OptionsMod) || @eval import Options
```

Comment:
Ok it's probably safer to use `@eval` (actually in https://github.com/JuliaLang/julia/issues/12233 Jeff later suggested `eval(Main, :(import Options))`)  even though I'm not sure exacly what would be different in this case.

About Options being necessary, see https://github.com/carlobaldassi/ArgParse.jl/issues/13

The new warning is indeed annoying, this issue is https://github.com/JuliaLang/Options.jl/issues/4.

Compat is needed for `startswith` and other changes, see e.g. https://github.com/carlobaldassi/ArgParse.jl/commit/996d7be91142731edc07ab10eead324201cf936f